### PR TITLE
Add e2e test for onboarding modal

### PR DESCRIPTION
**Pull request by jeffin143**
Resolve https://github.com/opencollective/opencollective/issues/3116

# Description

Add test for onboarding modal
# Screenshots

### CODE REVIEW

File: test/cypress/integration/38-onboarding-modal.test.js [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
+    cy.get('[data-cy="custom-checkbox"]').click();
+    cy.wait(300);
+    cy.get('button[type="submit"]').click();
+    cy.wait(1000);
+    // check if there is Modal with New collective created message
+    cy.get('[data-cy="onboarding-collective-created"]').contains('The New collective Collective has been created!');
+    cy.get('[data-cy="step-forward-button"]').click();
+    // Current count of admins
+    cy.get('[data-cy="profile-card"]').children().should('have.length', 1);
+    cy.get('[data-cy="admin-picker"]').type('test user{enter}');
```
Comment (jeffin143) [inline test/cypress/integration/38-onboarding-modal.test.js:21]: `cy.get().type()` doesn't work here, because after searching for user, you have to hit enter, Not sure why this line works sometimes and throws error other time. Couldn't find docs for this. Would greatly appreciate any help

Comment (jeffin143) [reply test/cypress/integration/38-onboarding-modal.test.js]: This is resolved

**Comment by sbinlondon** [comment]
Thanks for your work so far @jeffin143 I will be sure to have a look at it today.

**Comment by sbinlondon** [comment]
Looks good overall!

Some feedback:

I don't think we want to set the environment variable for `ONBOARDING_MODAL` to true in `env.js` and `next.config.js`. We can set env vars just for Cypress a few different ways though. I think maybe option 1 would be most relevant [here](https://docs.cypress.io/guides/guides/environment-variables.html#Option-1-configuration-file) if you want to give it a try.

Otherwise, great best practices using `data-cy` 🙌  Have suggested two small changes to remove commented out lines.

File: test/cypress/integration/38-onboarding-modal.test.js [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+    cy.get('[data-cy="custom-checkbox"]').click();
+    cy.wait(300);
+    cy.get('button[type="submit"]').click();
+    cy.wait(1000);
+    // check if there is Modal with New collective created message
+    cy.get('[data-cy="onboarding-collective-created"]').contains('The New collective Collective has been created!');
+    cy.get('[data-cy="step-forward-button"]').click();
+    // Current count of admins
+    cy.get('[data-cy="profile-card"]').children().should('have.length', 1);
+    // cy.get('[data-cy="admin-picker"]').type('test user{enter}');
```
Comment (sbinlondon) [inline test/cypress/integration/38-onboarding-modal.test.js:22]: Can we remove the commented out line here and on line 24?

File: components/onboarding-modal/OnboardingModal.js [hunk 3]
```diff
@@ -258,7 +258,7 @@ class OnboardingModal extends React.Component {
     const { step, isSubmitting, error, noOverlay } = this.state;
 
     return (
-      <React.Fragment>
+      <React.Fragment data-cy="collective-modal">
```
Comment (Betree) [inline components/onboarding-modal/OnboardingModal.js:261]: `Fragment`s are not included in the DOM, is this `data-cy` intended?

File: env.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ const defaults = {
   RECAPTCHA_SITE_KEY: '6LcyeXoUAAAAAFtdHDZfsxncFUkD9NqydqbIFcCK',
   RECAPTCHA_ENABLED: false,
   CLIENT_ANALYTICS_ENABLED: false,
-  ONBOARDING_MODAL: false,
```
Comment (Betree) [inline env.js:21]: Do we really want to change the ENV variables? Activating the onboarding modal by default seems out of scope

File: test/cypress/integration/25-host.apply.test.js [hunk 5]
```diff
@@ -18,8 +18,6 @@ describe('apply to host', () => {
     cy.get('button[type="submit"]').click();
     cy.wait(1000);
     cy.get('[data-cy="collective-title"]', { timeout: 10000 }).contains('New collective');
-    cy.get('.NotificationBar h1').contains('success');
-    cy.get('.NotificationBar p').contains('BrusselsTogether ASBL');
```
Comment (Betree) [inline test/cypress/integration/25-host.apply.test.js:22]: Why removing that?

File: test/cypress/integration/38-onboarding-modal.test.js [hunk 6]
```diff
@@ -1,17 +1,53 @@
...
-  it.skip('Adds admins to the list when you invite them', () => {
-    // test here
-  });
+import { randomEmail, randomSlug } from '../support/faker';
 
-  it.skip('Removes admins from the list when you click x to remove them', () => {
-    // test here
+describe('Onboarding modal', () => {
+  before(() => {
+    (process.env.ONBOARDING_MODAL = true), cy.login({ redirect: '/create/community' });
```
Comment (Betree) [inline test/cypress/integration/38-onboarding-modal.test.js:5]: This is an unusual syntax, we should use something more standard.

I'm also not sure if setting `ONBOARDING_MODAL` here has any effect, Cypress and frontend are supposed to have two different processes with separate Javascript contexts.

File: test/cypress/integration/25-host.apply.test.js (hunk 5 above)
Comment (jeffin143) [reply test/cypress/integration/25-host.apply.test.js:22]: Once Modal is enabled, Notification bar is not there 

It's either Modal or Notification bar : https://github.com/opencollective/opencollective-frontend/blob/5376b7543a0c97f8689ad47792273cd35a772e37/components/create-collective/index.js#L83

File: components/onboarding-modal/OnboardingModal.js (hunk 3 above)
Comment (jeffin143) [reply components/onboarding-modal/OnboardingModal.js:261]: You are right, That was not intended, Removed

File: test/cypress/integration/38-onboarding-modal.test.js (hunk 6 above)
Comment (jeffin143) [reply test/cypress/integration/38-onboarding-modal.test.js:5]: Since Now it is default we need not set it, Removed as well

File: test/cypress/integration/38-onboarding-modal.test.js (hunk 2 above)
Comment (jeffin143) [reply test/cypress/integration/38-onboarding-modal.test.js]: Removed

File: env.js (hunk 4 above)
Comment (jeffin143) [reply env.js:21]: Replied on Slack

Comment (sbinlondon) [reply env.js:21]: @znarf had said to @jeffin143 we can go ahead and make it the default and remove the flags https://opencollective.slack.com/archives/C0RMV6F8C/p1589207632341300?thread_ts=1589206808.339800&cid=C0RMV6F8C

File: test/cypress/integration/38-onboarding-modal.test.js [hunk 7]
```diff
@@ -1,17 +1,51 @@
...
+      cy.wrap($form).find('input[name="email"]').type(invitedUserEmail);
+      cy.wrap($form).find('input[name="name"]').type('AmazingNewUser');
+      cy.wrap($form).find('button[type="submit"]').click();
+    });
+    // Profile card should have length 2 since test user is added
+    cy.get('[data-cy="profile-card"]').children().should('have.length', 2);
+    cy.get('[data-cy="name-of-admins"]').contains('AmazingNewUser');
+    // Delete test user
+    cy.get('[data-cy="remove-user"] > button').click();
+    // Again the length shoudl be 1
```
Comment (jamesgeorge007) [inline test/cypress/integration/38-onboarding-modal.test.js:34]:
```suggestion
    // Again the length should be 1
```

File: test/cypress/integration/38-onboarding-modal.test.js [hunk 8]
```diff
@@ -1,17 +1,51 @@
...
+    // Add Github, Twitter, and website links
+    cy.get(`input[name="website"]`).type('opencollective.com/testCollective');
+    cy.get(`input[name="twitterHandle"]`).type('testCollective');
+    cy.get(`input[name="githubHandle"]`).type('testCollective');
+    // Finish creating collective
+    cy.get('[data-cy="finish-button"]').click();
+    cy.get('[data-cy="welcome-collective"]').contains('Welcome to your new Collective!');
+    // close the modal
+    cy.get('[data-cy="close-button"]').click();
+    // Check if the llink have been added
```
Comment (jamesgeorge007) [inline test/cypress/integration/38-onboarding-modal.test.js:46]:
```suggestion
    // Check if the link have been added
```

**Comment by sbinlondon** [comment]
Thanks for your work @jeffin143 - I'll squash and merge this 🙏
